airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/strategies/offset_increment.py

```diff
@@ -55,7 +63,10 @@ def reset(self):
         self._offset = 0
 
     def get_page_size(self) -> Optional[int]:
-        page_size = self._page_size.eval(self.config)
-        if not isinstance(page_size, int):
-            raise Exception(f"{page_size} is of type {type(page_size)}. Expected {int}")
-        return page_size
+        if self._page_size:
+            page_size = self._page_size.eval(self.config)
```

Comment:
this won't work if the page size interpolated string references `response`, right?

Reply:
That is correct. Although I'm not sure how we can do this because this part of the code doesn't have access to the response object the same way the `next_page_token()` method does. It also is not super critical either because if an API doesn't support sending page_size (like Discourse), it wouldn't matter if we could get a page_size from the response because the API wouldn't utilize it anyway.

The unfortunate part is the inconsistency as to what interpolation contexts the strategy has access and it shouldn't be something the developer should need to worry about. But that being said, its a pretty niche case for APIs that don't need this parameter since we went almost a year before finding one. I'm not sure that a wider refactor of our request options methods is a critical need to address this.